OpenRA.Mods.Common/Warheads/Warhead.cs

```diff
@@ -16,30 +16,28 @@
 
 namespace OpenRA.Mods.Common.Warheads
 {
-	public enum ImpactType
+	public enum ImpactActorType
 	{
 		None,
-		Ground,
-		Air,
-		TargetHit
-	}
-
-	public enum ImpactTargetType
-	{
-		NoActor,
-		ValidActor,
-		InvalidActor
+		Invalid,
+		Valid,
 	}
 
 	[Desc("Base warhead class. This can be used to derive other warheads from.")]
 	public abstract class Warhead : IWarhead
 	{
-		[Desc("What types of targets are affected.")]
+		[Desc("What types of actor targets are affected.")]
 		public readonly BitSet<TargetableType> ValidTargets = new BitSet<TargetableType>("Ground", "Water");
 
-		[Desc("What types of targets are unaffected.", "Overrules ValidTargets.")]
+		[Desc("What types of actor targets are unaffected.", "Overrules ValidTargets.")]
 		public readonly BitSet<TargetableType> InvalidTargets;
 
+		[Desc("What types of terrain targets are affected. Leave empty to affect all terrain not explicitly listed under InvalidTerrainTargets.")]
+		public readonly BitSet<TargetableType> ValidTerrainTargets;
```

Comment:
Are these actually needed anymore? #18068 seems to have done fine without them.

Reply:
Eh, looks like I made a rebase error or pushed the wrong branch. Yeah, I meant to drop those.